index.js

```diff
@@ -54,11 +55,23 @@ function decorateLayer(layer, keysToKeep, newProps) {
         keepLookup[keys.indexOf(keysToKeep[i])] = true;
     }
 
+    if (requiredKeys && requiredKeys.length) {
+        for (i = 0; i < requiredKeys.length; i++) {
+            var index = keys.indexOf(requiredKeys[i]);
+            if (index !== -1) requiredIndices.push(index);
+        }
+    }
```

Comment:
sure, makes sense
will also take a look at the bench script to confirm that we don't add too much computation